src/components/TodoPoint.jsx

```diff
@@ -1,33 +1,84 @@
 import React from 'react';
 
+
 class TodoPoint extends React.Component {
     constructor(props) {
         super(props);
-
         this.state = {
-            bIsImportant: false
+            bIsImportant: false,
+            bIsDone: false,
+            bIsInEdit: false
         };
         this.toggleImportant = this.toggleImportant.bind(this);
+        this.toggleDone = this.toggleDone.bind(this);
+    }
+
+    editTask = () => {
+        const newState = {
+            ...this.state,
+            bIsInEdit: true
+        }
+        this.setState(newState)
+    }
+
+    updateValue = (event) => {
+        const newText = event.target.value;
+        const oldText = event.target.defaultValue
+        if (event.key === 'Enter') {
+            const newState = {
+                ...this.state,
+                bIsInEdit: !this.state.bIsInEdit
+            }
+            this.props.changeText(oldText, newText)
+            this.setState(newState)
+        }
     }
 
     toggleImportant() {
+
         this.setState({
             ...this.state,
             bIsImportant: !this.state.bIsImportant
         });
+
+    }
+
+    toggleDone() {
+        this.setState(
+            {
+                ...this.state,
+                bIsDone: !this.state.bIsDone
+            }
+        )
+    }
+    getDoneStyle() {
+        return this.state.bIsDone ? "line-through" : ""
     }
 
     getTextColor() {
         return this.state.bIsImportant ? "red" : "black";
     }
 
-    render() {
-        const { removeTask, todoText } = this.props;
+    wichElement = (content) => {
+        const EditElem = <input onKeyDown={this.updateValue} type="text" defaultValue={content}></input>
+        const DefaultElem = <li style={{ textDecoration: this.getDoneStyle(), width: "150px", color: this.getTextColor() }} onClick={this.editTask}>
+            {content}</li>
+        return (this.state.bIsInEdit) ?
+            EditElem
+            :
+            DefaultElem
+    }
+
 
+
+    render() {
+        const { removeTask, todoText, todoDate } = this.props;
         return <div style={{ display: "flex", marginBottom: "5px" }}>
-            <li style={{ width: "150px", color: this.getTextColor() }}>{todoText}</li>
+            {this.wichElement(todoText)}
```

Comment:
this is not even proper English.

also cryptic naming in general. You could do smth like `getTaskTextElement()`.